### DISCUSSION TITLE
How do you check if a variable is an int or string?

### DISCUSSION THREAD

**Discussion by MrMidnightOfficial** [vlang/v#14902] [General]
How do you check if a variable is an int or string?

**Comment by JalonSolov** [comment]
I don't understand... it's your code, so you already know the variable type.

**Comment by Wajinn** [comment]
A simple way is to use `typeof`.  More complex ways would be to use sum types (type SI = int | string) or use it in a function's parameters where in the body of the function it uses `is` and checks (if foo is int).

```
fn main() {
	int_var := 3
	str_var := 'Z'
	bool_var := false
		
	println(typeof(int_var).name + '\n' + typeof(str_var).name + '\n' + typeof(bool_var).name)
}
```

**Reply by MrMidnightOfficial** [reply]
> A simple way is to use `typeof`. More complex ways would be to use sum types (type SI = int | string) or use it in a function's parameters where in the body of the function it uses `is` and checks (if foo is int).
> 
> ```
> fn main() {
> 	int_var := 3
> 	str_var := 'Z'
> 	bool_var := false
> 		
> 	println(typeof(int_var).name + '\n' + typeof(str_var).name + '\n' + typeof(bool_var).name)
> }
> ```

TYSM
